server/room.ts

```diff
@@ -15,6 +18,7 @@ export class Room {
     | { assignTime?: number; pass?: string; host?: string; id?: string }
     | undefined = undefined;
   private io: SocketIO.Server;
+  private socket?: Socket;
```

Comment:
This will change on each connection. I don't think you can have socket as a instance variable on the Room.

Reply:
Should we then maybe have a Connection class which handles one Connection each? I just realized we were passing around the socket variable a lot without the need to do so.

Reply:
Can you show an example of how that would look?

Reply:
I added the Connection class to my PR, see here:
https://github.com/howardchung/watchparty/pull/27/files#diff-1690407ef0505b9c0e8f0f9a529a8e4a

Reply:
This means on the actual connection we just have to create a new connection object:
https://github.com/howardchung/watchparty/pull/27/files#diff-cbe0f6ed81af545b15a7b13f43b39758R39